src/bp/nlu/stan/model-repo.ts

```diff
@@ -1,31 +1,90 @@
 import crypto from 'crypto'
 import fse, { WriteStream } from 'fs-extra'
 import _ from 'lodash'
-import mkdirp from 'mkdirp'
 import * as NLUEngine from 'nlu/engine'
 import path from 'path'
+import Logger from 'simple-logger'
 import { Stream } from 'stream'
 import tar from 'tar'
 import tmp from 'tmp'
+import {
+  Database,
+  DBStorageDriver,
+  DiskStorageDriver,
+  GhostService,
+  ScopedGhostService,
+  MemoryObjectCache
+} from './simple-ghost'
+interface FSDriver {
+  driver: 'fs'
+}
+
+interface DBDriver {
+  driver: 'db'
+  dbURL: string
+}
+
+export type ModelRepoOptions = (FSDriver | DBDriver) & {
```

Comment:
You did this because you want to let the compiler know that if `driver === 'db'` then dbURL is a defined property right? If so: 

Noooooice

Reply:
Yes